### PR TITLE
Accommodate breaking changes for geometry refactoring

### PR DESCRIPTION
**Pull request by knoepfel**
Accommodate changes from:

- https://github.com/LArSoft/larcoreobj/pull/19
- https://github.com/LArSoft/larcorealg/pull/37

----

**N.B.**: This PR should not be merged until the above PRs have been merged into a LArSoft release.

### CODE REVIEW

File: duneprototypes/Protodune/singlephase/DetectorServices/Providers/DetectorPropertiesProtoDUNEsp.cxx [hunk 1]
```diff
@@ -340,7 +335,7 @@ namespace spdp{
     // Calculate kinematic quantities.
 
     double bg = mom / mass;           // beta*gamma.
-    double gamma = sqrt(1. + bg*bg);  // gamma.
+    double gamma = square(1. + bg*bg);  // gamma.
```
Comment (aolivier23) [inline duneprototypes/Protodune/singlephase/DetectorServices/Providers/DetectorPropertiesProtoDUNEsp.cxx:338]: Are you sure we're not introducing a math mistake here?

**Review by aolivier23 — CHANGES_REQUESTED** [review]
All looks good except the potential math mistake in my comment.  We changed a sqrt() to a sqr().  I think it should remain a sqrt().  Can you double-check my work on the math?

File: duneprototypes/Protodune/singlephase/DetectorServices/Providers/DetectorPropertiesProtoDUNEsp.cxx (hunk 1 above)
Comment (knoepfel) [reply duneprototypes/Protodune/singlephase/DetectorServices/Providers/DetectorPropertiesProtoDUNEsp.cxx:338]: Good catch, @aolivier23!  Yes, this change is a mistake.  Are you able to revert it?

**Comment by knoepfel** [comment]
> All looks good except the potential math mistake in my comment. We changed a sqrt() to a sqr(). I think it should remain a sqrt(). Can you double-check my work on the math?

You are correct, @aolivier23.  It should be `sqrt()`.  Sorry about that.

**Comment by aolivier23** [comment]
Thank you for confirming.  I will revert that one line to sqrt() before merging after lunch.  Everything else looks good.

**Review by aolivier23 — APPROVED** [review]
Approved now that math is correct.  Verified that everything else is just a transition to the new geometry interface.

**Pull request merged**